src/main/java/nl/ordina/jobcrawler/service/VacancyService.java

```diff
@@ -39,9 +44,12 @@ public Vacancy add(Vacancy vacancy) {
 
     public Vacancy doesRecordExist(String url) {
```

Comment:
Name implies a boolean return type. getExistingRecord would align more with the implementation. 
Also, consider returning an Optional instead of possible null. 
The exception is fine though, because that indicates an inconsistent state of the system, and the system should fail because of that.

Reply:
Agree. Tried to change it to an Optional, see commit https://github.com/mostafa-221/jobcrawler-backend/pull/2/commits/2ce22091512de8a7c504d24293e59c9caca2f244
Might need to look into the syntax as that can probably be improved.